tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -161,13 +167,57 @@ def resources(self, query=None):
                     local_session(self.session_factory).get_default_project())
                 return []
             raise
-        self._cache.save(key, resources)
-        return self.filter_resources(resources)
 
     def augment(self, resources):
         return resources
 
 
+class ChildResourceManager(QueryResourceManager):
+
+    @classmethod
+    def get_parent_annotation_key(cls):
+        parent_resource_type = cls.resource_type.parent[0]
+        return 'c7n:{}'.format(parent_resource_type)
+
+    def get_resource(self, resource_info):
+        parent_resource_info = self._get_parent_resource_info(resource_info)
```

Comment:
do the super call upfront, and then have the child resource manager assemble the parent onto the resource, glomming into resource_info (which is event metadata) and just to have the type info get method extract it and stuff back into the resource dict is implicit and extraneous if its just done here. augmentation is a responsibility of the resource manager not the source, and the addition of the parent is just that.

Reply:
Agree. Thanks